org.adempiere.base/src/org/compiere/model/MLookup.java

```diff
@@ -993,6 +991,8 @@ public MLoader()
 		 */
 		protected void doRun()
 		{
+			/** Number of max rows to load	*/
+			int	MAX_ROWS = MSysConfig.getIntValue(MSysConfig.MAX_ROWS_IN_LOOKUP, 10000, Env.getAD_Client_ID(Env.getCtx()));

```

Comment:
I think we should still have a hard limit here (for e.g, 50000) since a high max rows value here is a potential source of performance disaster.

Reply:
> I think we should still have a hard limit here (for e.g, 50000) since a high max rows value here is a potential source of performance disaster.

I would suggest to do it adding a warning in the comment of the SysConfig, something like "WARNING: increasing this value affects heavily performance and memory, use it wisely"

WDYT?

Reply:
I would prefer a warning plus a hard limit. A wrong configuration would effectively kill a server, not just affecting performance.